src/alire/alire-test_runner.adb

```diff
@@ -254,23 +274,33 @@ package body Alire.Test_Runner is
       -- Append --
       ------------
 
-      procedure Append (Dir_Entry : Adirs.Directory_Entry_Type) is
-         --  Helper function to append all .adb files in a folder
+      procedure Append (This         : Den.Walk.Item;
+                        Unused_Enter : in out Boolean;
+                        Unused_Stop  : in out Boolean)
+      is
+         --  Helper function to append all .adb files in a tree
          --  to the `Test_List` vector
 
-         Name : constant String := Adirs.Simple_Name (Dir_Entry);
+         Name : constant String :=
+                  Strip_Prefix (This.Path,
+                                Prefix => Root.Path / "src");
       begin
          if Name'Length > 4
            and then Name (Name'Last - 3 .. Name'Last) = ".adb"
            and then (Filter.Is_Empty
                      or else (for some F of Filter
                               => Ada.Strings.Fixed.Index (Name, F) /= 0))
```

Comment:
Something I missed during the filtering initially: this also matches on the package name (`crate_test-...`) which is hidden during output :grimacing: I don't know if we should do something about that

Reply:
I noticed too. Probably yes just in case for some cases it brings in extra unwanted tests.